Samples/NetworkMediaStreamer/RTPMediaStreamer/src/RTPStreamSink.cpp

```diff
@@ -0,0 +1,443 @@
+    {
+        auto sep3 = destination.find("&", sep2);
+        if (sep3 != std::string::npos)
+        {
+            m_ssrc = std::stoi(destination.substr(sep2 + param.size(), sep3 - sep2));
+        }
+    }
+
+    if (!packetHandler)
+    {
+       param = "localrtpport=";
+        auto sep1 = destination.find(":");
+        auto ipaddr = destination.substr(0, sep1);
+        sep2 = destination.find(param);
+        m_remotePort = std::stoi(destination.substr(sep1 + 1, sep2 - sep1 - 1));
+        if (sep2 != std::string::npos)
+        {
+            auto sep3 = destination.find("&", sep2);
+            m_localRTPPort = std::stoi(destination.substr(sep2 + param.size(), sep3 - sep2));
+        }
+        else
+        {
+            m_localRTPPort = 6970;
+        }
+
+        sockaddr_in Server;
+        Server.sin_family = AF_INET;
+        Server.sin_addr.s_addr = INADDR_ANY;
+        for (u_short P = m_localRTPPort; P < 0xFFFE; P += 2)
+        {
+            m_RtpSocket = socket(AF_INET, SOCK_DGRAM, 0);
+            Server.sin_port = htons(P);
+            if (bind(m_RtpSocket, (sockaddr*)&Server, sizeof(Server)) == 0)
+            {   // Rtp socket was bound successfully. Lets try to bind the consecutive Rtsp socket
+                m_RtcpSocket = socket(AF_INET, SOCK_DGRAM, 0);
+                Server.sin_port = htons(P + 1);
+                if (bind(m_RtcpSocket, (sockaddr*)&Server, sizeof(Server)) == 0)
+                {
+                    m_localRTPPort = P;
+                    m_localRTCPPort = P + 1;
+                    break;
+                }
+                else
+                {
+                    closesocket(m_RtpSocket);
+                    m_RtpSocket = INVALID_SOCKET;
+                    closesocket(m_RtcpSocket);
+                    m_RtcpSocket = INVALID_SOCKET;
+                }
+            }
+            else
+            {
+                closesocket(m_RtpSocket);
+                m_RtpSocket = INVALID_SOCKET;
+            }
+        }
+
+        inet_pton(AF_INET, destination.c_str(), &(m_remoteAddr.sin_addr));
+        m_remoteAddr.sin_port = htons(m_remotePort);
+        m_remoteAddr.sin_family = AF_INET;
+    }
+}
+
+TxContext::~TxContext()
+{
+    if (m_RtpSocket != INVALID_SOCKET)
+    {
+        closesocket(m_RtpSocket);
+    }
+    if (m_RtcpSocket != INVALID_SOCKET)
+    {
+        closesocket(m_RtcpSocket);
+    }
+}
+
+void TxContext::SendPacket(winrt::Windows::Storage::Streams::IBuffer buf)
+{
+    if (m_packetHandler)
+    {
+        m_packetHandler(nullptr, buf);
+    }
+    else
+    {
+        auto pBuf = buf.data();
+        auto sz = buf.Length();
+        sendto(m_RtpSocket, (char*)pBuf, (int)sz, 0, (SOCKADDR*)&m_remoteAddr, sizeof(m_remoteAddr));
+    }
+    m_SequenceNumber++;
+}
+
+
+RTPVideoStreamSink::RTPVideoStreamSink(IMFMediaType* pMediaType, IMFMediaSink* pParent, DWORD dwStreamID)
+    : NwMediaStreamSinkBase(pMediaType, pParent, dwStreamID)
+    , m_packetizationMode(1)

```

Comment:
>m_packetizationMode(1) [](start = 6, length = 22)

is m_packetizationMode ever == 0 ?

Reply:
in the future yes. there is code that supports it, but the plumbing of this through the params is not enabled right now.

---
In reply to: [492968011](https://github.com/microsoft/Windows-Camera/pull/13#discussion_r492968011) [](ancestors = 492968011)

Reply:
ok, maybe add a todo comment to address that